borg/hashindex.pyx

```diff
@@ -175,9 +192,39 @@ cdef class ChunkIndex(IndexBase):
 
     def __contains__(self, key):
         assert len(key) == self.key_size
-        data = <int *>hashindex_get(self.index, <char *>key)
+        data = <int32_t *>hashindex_get(self.index, <char *>key)
         return data != NULL
 
+    def incref(self, key):
+        """Increase refcount for 'key', return (refcount, size, csize)"""
+        assert len(key) == self.key_size
+        data = <int32_t *>hashindex_get(self.index, <char *>key)
+        if not data:
+            raise KeyError(key)
+        cdef int32_t refcount = _le32toh(data[0])
+        if refcount != INT32_MAX:
+            if INT32_MAX - refcount < 1:
```

Comment:
how does the if above protects us here?

refcount = -2;
if refcount != INT32_MAX then // True
  refcount--;
now refcount is -3, where as it should become INT32_MAX thereby repairing earlier overflow before this code was introduced (or just plain cache corruption of some other sort).